packages/components/src/color-picker/alpha.js

```diff
@@ -0,0 +1,142 @@
+	}
+
+	componentDidMount() {
+		this.mousetrap = new Mousetrap( this.container );
+		this.mousetrap.bind( 'up', () => this.increase() );
+		this.mousetrap.bind( 'right', () => this.increase() );
+		this.mousetrap.bind( 'shift+up', () => this.increase( 0.1 ) );
+		this.mousetrap.bind( 'shift+right', () => this.increase( 0.1 ) );
+		this.mousetrap.bind( 'pageup', () => this.increase( 0.1 ) );
+		this.mousetrap.bind( 'end', () => this.increase( 1 ) );
+		this.mousetrap.bind( 'down', () => this.decrease() );
+		this.mousetrap.bind( 'left', () => this.decrease() );
+		this.mousetrap.bind( 'shift+down', () => this.decrease( 0.1 ) );
+		this.mousetrap.bind( 'shift+left', () => this.decrease( 0.1 ) );
+		this.mousetrap.bind( 'pagedown', () => this.decrease( 0.1 ) );
+		this.mousetrap.bind( 'home', () => this.decrease( 1 ) );
+	}
+
+	componentWillUnmount() {
+		this.unbindEventListeners();
+		this.mousetrap.reset();
+	}
+
+	increase( amount = 0.01 ) {
+		const { hsl, onChange = noop } = this.props;
+		amount = parseInt( amount * 100, 10 );
+		const change = {
+			h: hsl.h,
+			s: hsl.s,
+			l: hsl.l,
+			a: ( parseInt( hsl.a * 100, 10 ) + amount ) / 100,
+			source: 'rgb',
+		};
+		onChange( change );
+	}
+
+	decrease( amount = 0.01 ) {
+		const { hsl, onChange = noop } = this.props;
+		const change = {
+			h: hsl.h,
+			s: hsl.s,
+			l: hsl.l,
+			a:
+				hsl.a <= amount ?
+					0 :
+					( parseInt( hsl.a * 100, 10 ) - parseInt( amount * 100, 10 ) ) / 100,
+			source: 'rgb',
+		};
+		onChange( change );
+	}
+
+	handleChange( e ) {
+		const { onChange = noop } = this.props;
+		const change = calculateAlphaChange( e, this.props, this.container );
+		if ( change ) {
+			onChange( change, e );
+		}
+	}
+
+	handleMouseDown( e ) {
+		this.handleChange( e );
+		window.addEventListener( 'mousemove', this.handleChange );
+		window.addEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	handleMouseUp() {
+		this.unbindEventListeners();
+	}
+
+	unbindEventListeners() {
+		window.removeEventListener( 'mousemove', this.handleChange );
+		window.removeEventListener( 'mouseup', this.handleMouseUp );
+	}
+
+	render() {
+		const { rgb } = this.props;
+		const rgbString = `${ rgb.r },${ rgb.g },${ rgb.b }`;
+		const gradient = {
+			background: `linear-gradient(to right, rgba(${ rgbString }, 0) 0%, rgba(${ rgbString }, 1) 100%)`,
+		};
+
+		const pointerLocation = { left: `${ rgb.a * 100 }%` };
+
+		return (
+			<div className="components-color-picker__alpha">
+				<div
+					className="components-color-picker__alpha-gradient"
+					style={ gradient }
+				/>
+				{ /* eslint-disable jsx-a11y/no-static-element-interactions */ }
+				<div
+					className="components-color-picker__alpha-bar"
+					ref={ ( container ) => ( this.container = container ) }
+					onMouseDown={ this.handleMouseDown }
+					onTouchMove={ this.handleChange }
+					onTouchStart={ this.handleChange }>
+					<button
+						role="slider"
+						aria-valuemax="1"
```

Comment:
Not sure a button is a proper element for a `role="slider"`. We're basically overriding the native semantics of the button and it's against the W3C recommendations. See https://www.w3.org/TR/using-aria/#notes2

Also, the ARIA [slider implementation details](https://www.w3.org/TR/wai-aria-practices-1.1/#slider) and the examples ([example 1](https://www.w3.org/TR/wai-aria-practices-1.1/examples/slider/slider-1.html) and [example 2](https://www.w3.org/TR/wai-aria-practices-1.1/examples/slider/slider-2.html)) simply use a focusable div. The proper semantics is given by the ARIA role. This applies also to the other slider.

Reply:
I needed to use a natively focusable element like a button for a screen reader (I think this was VoiceOver) to properly pick up on key events - but that issue is fixed with the key events fix in 5fddf98af8d9cffa3b85b705356ef6427e0cdd7b, so I've updated this to a focusable div in 1ea4e2a3486e103a72e6b05a5ae32fd5b34a48a0.